skrobot/planner/utils.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+
+from skrobot.coordinates import CascadedCoords
+from skrobot.coordinates import Coordinates
+from skrobot.coordinates.math import rpy_angle
+from skrobot.coordinates.math import rpy_matrix
+
+
+def set_robot_config(robot_model, joint_list, av, with_base=False):
+    """A utility function for setting robot state
+
+    Parameters
+    ----------
+    robot_model : skrobot.model.CascadedLink
+        robot model
+    joint_list : list[skrobot.model.Joint]
+        joint list to be set
+    av : numpy.ndarray[float](n_dof,)
+        angle vector which has n_dof dims.
+    with_base : bool
+        If `with_base=False`, `n_dof` is the number of joints `n_joint`,
+        but if `with_base=True`, `n_dof = n_joint + 3`.
+
+    Returns
+    -------
+    """
+
+    if with_base:
+        assert len(joint_list) + 3 == len(av)
+    else:
+        assert len(joint_list) == len(av)
+
+    if with_base:
+        av_joint, av_base = av[:-3], av[-3:]
+        x, y, theta = av_base
+        co = Coordinates(pos=[x, y, 0.0], rot=rpy_matrix(theta, 0.0, 0.0))
+        robot_model.newcoords(co)
+    else:
+        av_joint = av
+
+    for joint, angle in zip(joint_list, av_joint):
+        joint.joint_angle(angle)
+
+
+def get_robot_config(robot_model, joint_list, with_base=False):
+    """A utility function for getting robot state
+
+    Parameters
+    ----------
+    robot_model : skrobot.model.CascadedLink
+        robot model
+    joint_list : list[Joint]
+        joint list of which you want to know the angles
+    with_base : bool
+        If set to `True`, base position is also computed.
+
+    Returns
+    -------
+    av_whole (or av_joint) : numpy.array(n_dof,)
+        angle vector. If `with_base=False`, `n_dof` is the number of
+        joints `n_joint`, but if `with_base=True`, `n_dof = n_joint + 3`.
+    """
+    av_joint = np.array([j.joint_angle() for j in joint_list])
+    if with_base:
+        x, y, _ = robot_model.translation
+        rpy = rpy_angle(robot_model.rotation)[0]
+        theta = rpy[0]
+        av_whole = np.hstack((av_joint, [x, y, theta]))
+        return av_whole
+    else:
+        return av_joint
+
+
+def forward_kinematics_multi(robot_model,
+                             joint_list,
+                             av,
+                             move_target_list,
+                             with_rot,
+                             with_base,
+                             with_jacobian=True):
+    """Compute fk for multiple feature points
+
+    Parameters
+    ----------
+    robot_model : skrobot.model.CascadedLink
+        robot model.
+    joint_list : list[skrobot.model.Joint]
+        joint to be controlled
+    av : numpy.ndarray(n_dof,)
+        angle vector.
+    move_target_list : list[skrobot.coordinates.CascadedCoords]
+        the list has `n_feature` elements. Each element is the
+        coordinate of the features points.
+    with_rot : bool
```

Comment:
I think that `with_rot` can be replaced with `rotation_axis.`
It would be useful.
I would like to hear your opinion.

Reply:
Thanks for your advise. From the knowledge of the euslisp, I guess specifying `rotation_axis` is for freeing some rotation_axis constraints. For example rotation_axis=`x` enables to compute jacobian for only about `x` axis. 

However in the planning, I currently don't think this kind of constraint-freeing will be used. So, I think `with_rot` is enough and it's more explicit. 

Aside from this PR, from the experience of euslisp as a beginner 1 year ago, I personally think all `rotation_axis` in `scikit-robot` should be divided into two parameters `with_rot` and `rotation_axis`, because just they are more explicit.

Reply:
OK. We think again when someone needs the feature in the future.